CenterForOpenScience/osf.io
# [ENG-2201] Registries Moderation Notifications

## Purpose

Add notifications to registries moderation machine

## Changes

- Makes the preprint moderation email templates generic to registrations.
- Makes Preprint provider moderation notifications a sub-class of both  PreprintProvider and RegistrationProvider
- Makes adding a moderator to a provider add notification subscriptions for withdraw requests
- Adds many tests.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify emails are sent at appropriate times
- Verify emails are grammatically correct
- Verify no additional error messages appear during notification actions.

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation

No new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2201
